src/redis-pubsub.ts

```diff
@@ -169,17 +175,23 @@ export class RedisPubSub implements PubSubEngine {
   private readonly subsRefsMap: Map<string, Set<number>>;
   private currentSubscriptionId: number;
 
-  private onMessage(pattern: string, channel: string, message: string) {
+  private onMessage(pattern: string, channel: string | Buffer, message: string | Buffer) {
+    if(typeof channel === 'object') channel = channel.toString('utf8');
+
     const subscribers = this.subsRefsMap.get(pattern || channel);
 
     // Don't work for nothing..
     if (!subscribers?.size) return;
 
     let parsedMessage;
     try {
-      parsedMessage = this.deserializer
-        ? this.deserializer(message, { pattern, channel })
-        : JSON.parse(message, this.reviver);
+      if(this.deserializer){
+        parsedMessage = this.deserializer(Buffer.from(message), { pattern, channel })
```

Comment:
The serializer/deserializer may not be dealing with buffers at all. I think the original code is correct. I think the only change needed is the check above for whether channel is an object.

Reply:
Yes, there are some serialization/desrialization libraries that don't necessarily output buffers or take in buffers. They could be encoding/decoding using strings (inefficient, I know). But then, I guess they would not be setting the eventName to messageBuffer in the config. So this is likely ok.